.github/workflows/gradle-build-and-upload.yml

```diff
@@ -0,0 +1,53 @@
+name: gradle-build-and-upload
+
+on:
+  workflow_call:
+    inputs:
+      release:
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  gradle_build_and_upload:
+    name: Gradle build and upload
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
```

Comment:
Just curious why we enforce a timeout here? Are we expecting flakiness?

Reply:
`gradle build` takes some time to finish, it solves all the dependencies and runs all the test.